src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -1453,6 +1453,134 @@ def test_aks_nodepool_get_upgrades(self, resource_group, resource_group_location
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_snapshot(self, resource_group, resource_group_location):
+        create_version, upgrade_version = self._get_versions(resource_group_location)
+        aks_name = self.create_random_name('cliakstest', 16)
+        aks_name2 = self.create_random_name('cliakstest', 16)
+        nodepool_name = self.create_random_name('c', 6)
+        nodepool_name2 = self.create_random_name('c', 6)
+        snapshot_name = self.create_random_name('s', 6)
+
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'aks_name2': aks_name2,
+            'location': resource_group_location,
+            'nodepool_name': nodepool_name,
+            'nodepool_name2': nodepool_name2,
+            'snapshot_name': snapshot_name,
+            'k8s_version': create_version,
+            'upgrade_k8s_version': upgrade_version,
+            'ssh_key_value': self.generate_ssh_keys()
+        })
+
+        # create an aks cluster not using snapshot
+        create_cmd = 'aks create --resource-group {resource_group} --name {name} --location {location} ' \
+                     '--nodepool-name {nodepool_name} ' \
+                     '--node-count 1 ' \
+                     '--ssh-key-value={ssh_key_value} -o json'
+        response = self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded')
+        ]).get_output_in_json()
+
+        cluster_resource_id = response["id"]
+        assert cluster_resource_id is not None
+        nodepool_resource_id = cluster_resource_id + "/agentPools/" + nodepool_name
+        self.kwargs.update({
+            'nodepool_resource_id': nodepool_resource_id,
+        })
+        print("The nodepool resource id %s " % nodepool_resource_id)
+
+        # create snapshot from the nodepool
+        create_snapshot_cmd = 'aks snapshot create --resource-group {resource_group} --name {snapshot_name} --location {location} ' \
+                              '--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/SnapshotPreview ' \
+                              '--nodepool-id {nodepool_resource_id} -o json'
+        response = self.cmd(create_snapshot_cmd, checks=[
```

Comment:
I have queued a live test for this case (sorry for the inconvenience, the pipeline cannot triggered by PR since last Friday, I am still looking for a fix) and I found it fails at "create snapshot from the nodepool" ([error log](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=48495186&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=a26fe913-e2be-5062-b08e-d15f1acc7ea2&l=847)).

Is this expected due to release has not been rolled out or we need to use/test this command in some specific region/sub? The test is performed on westcentralus and a sub with only `AKSHTTPCustomFeatures` feature registered.

Reply:
RP v20211028 release is in progress, I'm enabling snapshots related toggles in public regions gradually (https://msazure.visualstudio.com/DefaultCollection/CloudNativeCompute/_git/aks-rp/pullrequest/5292586, PR to enable them in Canary was checked in just now, but not deployed yet) as the RP release moves. It is possible we don't include snapshot related in Live tests for now?

Reply:
Sure, I would appreciate if you could add the test case name to file [ext_matrix_default.json](https://github.com/chengliangli0918/azure-cli-extensions/blob/charlili/snapshotcmd/src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json).

Reply:
test_aks_snapshot  was added to file ext_matrix_default.json